roles/matrix-synapse/tasks/workers/setup_install.yml

```diff
@@ -0,0 +1,100 @@
+---
+
+- name: Download synapse workers doc
+  local_action:
+    module: get_url
+    url: https://github.com/matrix-org/synapse/raw/master/docs/workers.md
+    dest: "{{ role_path }}/files/workers.upstream-documentation.md"
+  vars:
+    ansible_become: no
+
+- name: Download synapse workers doc and convert into YAML
+  local_action:
+    module: shell
+    cmd: "awk -f '{{ role_path }}/files/workers-doc-to-yaml.awk' -- '{{ role_path }}/files/workers.upstream-documentation.md' > '{{ role_path }}/vars/workers.yml'"
+    creates: "{{ role_path }}/vars/workers.yml"
+  vars:
+    ansible_become: no
+
+- name: Load list of available worker apps and endpoints
+  include_vars: "{{ role_path }}/vars/workers.yml"
```

Comment:
I'd suggest `delegate`ing the previous three tasks to localhost (and `set_fact`ing the vars after the extraction+conversion), or even going so far as to hard-code the endpoints in the playbook and maintain them.

Reply:
Ok admittedly, I'm an ansible n0ob so it would be nice if you could explain to me what difference it would make (for the first option).
For the latter, that would certainly be feasible. What advantage would it have though? :thinking:

Reply:
The advantage is that the tasks get run on the host used to deploy the server, i.e. the computer you are sitting infront of. Delegating a task like "download something from github" to localhost allows the download to succeed, even if the actual server might have a firewall which does not allow arbitrary egress traffic (i.e. blocking github (or other sites, in a corporate setting))